src/frontend/src/handler/create_source.rs

```diff
@@ -255,10 +259,46 @@ fn check_and_add_timestamp_column(
     }
 }
 
+fn bind_source_watermark(
+    session: &SessionImpl,
+    name: String,
+    source_watermarks: Vec<SourceWatermark>,
+    column_catalogs: &[ColumnCatalog],
+) -> Result<Vec<WatermarkDesc>> {
+    let mut binder = Binder::new(session);
+    binder.bind_column_defs(name.clone(), column_catalogs.to_vec())?;
+
+    let watermark_descs = source_watermarks
+        .into_iter()
+        .map(|source_watermark| {
+            let col_name = source_watermark.column.real_value();
+            let watermark_idx = binder.get_column_binding_index(name.clone(), &col_name)?;
+
+            let expr = binder.bind_expr(source_watermark.expr)?.to_expr_proto();
+
+            Ok::<_, RwError>(WatermarkDesc {
+                watermark_idx: Some(ProstColumnIndex {
+                    index: watermark_idx as u64,
+                }),
+                expr: Some(expr),
+            })
+        })
+        .try_collect()?;
+    Ok(watermark_descs)
+}
```

Comment:
Maybe it is too heavy here to construct a new binder. we can extract some bind expression logic later.

Reply:
Maybe in later PR